lightwood/encoder/categorical/onehot.py

```diff
@@ -1,140 +1,174 @@
 import torch
 import numpy as np
-from scipy.special import softmax
-from lightwood.encoder.text.helpers.rnn_helpers import Lang
 from lightwood.helpers.log import log
 from lightwood.encoder.base import BaseEncoder
+from lightwood.helpers.constants import _UNCOMMON_WORD
 
-UNCOMMON_WORD = '__mdb_unknown_cat'
-UNCOMMON_TOKEN = 0
+from typing import Dict, List, Iterable
 
 
 class OneHotEncoder(BaseEncoder):
     """
+    Creates a one-hot encoding (OHE) of categorical data. This creates a vector where each individual dimension corresponds to a category. A category has a 1:1 mapping between dimension indicated by a "1" in that position.
 
-    Why are we handling target weighting inside encoders? Simple: we'd otherwise have to compute per-index weighting inside the mixers, rather than having that code unified inside 2x encoders. So moving this to the mixer will still involve having to pass the target encoder to the mixer, but will add the additional complexity of having to pass a weighting map to the mixer and adding class-to-index translation boilerplate + weight setting for each mixer
-    """ # noqa
-    def __init__(self, is_target=False, target_weights=None, handle_unknown='unknown_token'):
-        super().__init__(is_target)
-        self._lang = None
-        self.rev_map = {}
+    OHE operates in 2 modes:
+        (1) "use_unknown=True": Makes an :math:`N+1` length vector for :math:`N` categories, the first index always corresponds to the unknown category.
```

Comment:
Ooh, `:math:` seems useful, there's a bunch of docstrings that could use this :eyes:

Reply:
:eyes: im probably not using it correctly sigh